### ISSUE TITLE
Decorators break _classCallCheck

### ISSUE THREAD

**Issue by matthewwithanm** [babel/babel#1189]
The following will throw "TypeError: Cannot call a class as a function":

```javascript
function factory(Ctor) {
  return () => new Ctor();
}

@factory
class K {
}

let k = K();
```

The reason is that the compiled code checks `instanceof` at construction time using a reference that has been replaced by the factory:

```javascript
var K = (function () {
  function K() {
    _classCallCheck(this, K);
  }

  K = factory(K) || K;
  return K;
})();
```

It seems like the late-bound check will need to use a different variable than the one that holds the decorator result.

**Comment by Dignifiedquire** [comment]
Already fixed in #1185

**Comment by sebmck** [comment]
I'll push out a patch now.

**Comment by sebmck** [comment]
Fixed as of 5.0.10, thanks!

**Comment by matthewwithanm** [comment]
Awesome, thanks!
